### PR TITLE
add rocketMq grayscale message plugin

### LINKED ISSUE #1539: support rocketMq grayscale message
### What would you like to be enhanced?

I want to enhance plugin to support rocketMq grayscale message.

### Why is this needed?

During gray deployment, gray instances need to consume or produce gray messages.

### PR DESCRIPTION
**Pull request by chengyouling**
What type of PR is this?

Feature.

What this PR does / why we need it?

During gray deployment, gray instances need to consume or produce gray messages.

Which issue(s) this PR fixes？

Fixes #1539

Does this PR introduce a user-facing change?

No

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-config/config/plugins.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ plugins:
   - service-visibility
   - tag-transmission
   - database-write-prohibition
+  - sermant-mq-grayscale
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-config/config/plugins.yaml]: The plug-in name should be mq-grayscale.

File: sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/Base.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+package io.sermant.mq.grayscale.config;
+
+/**
+ * base consumeType entry
+ *
+ * @author chengyouling
+ * @since 2024-05-27
+ **/
+public class Base {
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/Base.java]: Use specified name to name class

File: sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/CseMqGrayConfigHandler.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+     * @param event
+     */
+    public void handle(DynamicConfigEvent event) {
+        if ("grayscale.mq.config".equals(event.getKey())) {
+            if (event.getEventType() == DynamicConfigEventType.DELETE) {
+                MqGrayscaleConfigUtils.resetGrayscaleConfig();
+                return;
+            }
+            if (!StringUtils.isEmpty(event.getContent())) {
+                LOGGER.warning(String.format(Locale.ROOT,
```
Comment (daizhenyu) [inline sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/CseMqGrayConfigHandler.java]: not warning

**Review by daizhenyu — DISMISSED** [review]

File: sermant-agentcore/sermant-agentcore-config/config/plugins.yaml (hunk 1 above)
Comment (chengyouling) [reply sermant-agentcore/sermant-agentcore-config/config/plugins.yaml]: fixed

File: sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/Base.java (hunk 2 above)
Comment (chengyouling) [reply sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/Base.java]: fixed

File: sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/CseMqGrayConfigHandler.java (hunk 3 above)
Comment (chengyouling) [reply sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/config/CseMqGrayConfigHandler.java]: change info

**Pull request closed without merging**